Update Razor SDK to V6 (#22202)
dotnet/AspNetCore.Docs: aspnetcore/migration/50-to-60.md

+ docker pull mcr.microsoft.com/dotnet/aspnet:6.0
```

## Changes to the ASP.NET Core Razor SDK

The Razor compiler now leverages the new [source generators feature](https://devblogs.microsoft.com/dotnet/introducing-c-source-generators/) to generate compiled C# files from the Razor views and pages in a project. In previous versions:

* The compilation relied on the `RazorGenerate` and `RazorCompile` targets to produce the generated code. These targets are no longer valid. In .NET 6, both code generation and compilation are supported by a single call to the compiler. `RazorComponentGenerateDependsOn` is still supported to specify dependencies that are required before the build runs.
* A separate Razor assembly, `AppName.Views.dll`, was generated that contained the compiled view types in an application. This behavior has been deprecated and a single assembly `AppName.dll` is produced that contains both the app types and the generated views.

> * The app types in `AppName.Views.dll` were public. In .NET 6, the app types are in `AppName.dll` but are `internal sealed`. Apps doing type discover on `AppName.Views.dll` won't be able to do type discover on `AppName.dll`.

Author: Can you suggest some text to clarify what the delta is?

+ docker pull mcr.microsoft.com/dotnet/aspnet:6.0
```

## Changes to the ASP.NET Core Razor SDK

> The Razor compiler now leverages the new [source generators feature](https://devblogs.microsoft.com/dotnet/introducing-c-source-generators/) to generated compiled C# files from the Razor views and pages in a document. In previous versions:

Reviewer: ```suggestion
The Razor compiler now leverages the new [source generators feature](https://devblogs.microsoft.com/dotnet/introducing-c-source-generators/) to generate compiled C# files from the Razor views and pages in a project. In previous versions:
```

> * The compilation relied on an individual `RazorGenerate` and `RazorCompile` target to produce the generated code. These targets are no longer valid. In .NET 6, both code generation and compilation are supported by a single call to the compiler. `RazorComponentGenerateDependsOn` is still supported to specify dependencies that are required before the build runs.

Reviewer: ```suggestion
* The compilation relied on the `RazorGenerate` and `RazorCompile` targets to produce the generated code. These targets are no longer valid. In .NET 6, both code generation and compilation are supported by a single call to the compiler. `RazorComponentGenerateDependsOn` is still supported to specify dependencies that are required before the build runs.
```

* A separate Razor assembly, `AppName.Views.dll`, was generated that contained the compiled view types in an application. This behavior has been deprecated and a single assembly `AppName.dll` is produced that contains both the app types and the generated views.

> * The app types in `AppName.Views.dll` were public. In .NET 6, the app types are in `AppName.dll` but are not public. Apps doing type discover on `AppName.Views.dll` won't be able to do type discover on `AppName.dll`.

Reviewer: ```suggestion
* The app types in `AppName.Views.dll` were public. In .NET 6, the app types are in `AppName.dll` but are `internal sealed`. Apps doing type discover on `AppName.Views.dll` won't be able to do type discover on `AppName.dll`.
```

Might help to clarify what the delta is?

Make the following changes:

> * Remove all the targets listed under `Targets`.

Reviewer: ```suggestion
```

This isn't applicable for end-users. It reflects the changes made in sdk.md

> * Optional: Under `Properties` add the `EmitCompilerGeneratedFiles` property to the element. When `EmitCompilerGeneratedFiles` is set to `true`, the generated source files are written to disk. Setting to `true` is useful when debugging the compiler. The default is `false`.

Reviewer: ```suggestion
```

Ditto. This isn't a migration step.
